pkg/controller/infrastructureconfig/sync_vips_test.go

```diff
@@ -267,6 +267,19 @@ func Test_apiAndIngressVipsSynchronizer_VipsSynchronize(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "should not panic on empty PlatformStatus.VSphere field (VSphere UPI doesn't populate VSphere field)",
+			givenStatus: configv1.InfrastructureStatus{
+				Platform: configv1.VSpherePlatformType,
+				PlatformStatus: &configv1.PlatformStatus{
+					VSphere: nil,
+				},
+			},
+			wantStatus: configv1.InfrastructureStatus{
+				Platform:       configv1.VSpherePlatformType,
+				PlatformStatus: &configv1.PlatformStatus{},
```

Comment:
@creydr : here the given status is nil, so we shouldn't take any action right? your switch case matches explicitly on the `!= nil` part, so should want status after reconcile also be nil?

Reply:
Yes exactly. This test is only, to make sure we don't panic if `vsphere = nil` (what can happen in vsphere UPI). 
For the `!= nil` part, we have another test: https://github.com/openshift/cluster-network-operator/blob/1a56d16d64512522fbcbc95e0570f4a57ab14f11/pkg/controller/infrastructureconfig/sync_vips_test.go#L187-L208